tests/test_vocabulary.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from jschon import URIError, JSONSchema
+
+@pytest.mark.parametrize('draft', ('2019-09', '2020-12', 'next'))
+def test_non_normalized_id_allowed(draft):
+    schema_data = {
+        '$schema': f'https://json-schema.org/draft/{draft}/schema',
+        '$id': 'http://localhost:1234/foo/bar/../baz',
+    }
+    schema = JSONSchema(schema_data)
+    assert str(schema['$id'].value) == schema_data['$id']
+
+
+@pytest.mark.parametrize('draft', ('2019-09', '2020-12', 'next'))
+def test_nonempty_fragment_id_disallowed(draft):
+    schema_data = {
+        '$schema': f'https://json-schema.org/draft/{draft}/schema',
+        '$id': '#foo',
+    }
+    match = 'a fragment' if draft == 'next' else 'a non-empty fragment'
+    with pytest.raises(URIError, match=match):
+        schema = JSONSchema(schema_data)
```

Comment:
`schema` is not used so the assignment can be removed

Reply:
heh, I think this a leftover reaction to some linter config I had to work with that hated unassigned non-`None` results more than unused variables.  Weird! 🙃 thanks for noticing.